### PR TITLE
Add ability to rate limit via Validate class

### PR DESCRIPTION
**Pull request by samerton**

### CODE REVIEW

File: modules/Core/pages/login.php [hunk 1]
```diff
@@ -82,12 +86,14 @@
                 'email' => [
                     Validate::REQUIRED => $language->get('user', 'must_input_email'),
                     Validate::IS_BANNED => $language->get('user', 'account_banned'),
-                    Validate::IS_ACTIVE => $language->get('user', 'inactive_account')
+                    Validate::IS_ACTIVE => $language->get('user', 'inactive_account'),
+                    Validate::RATE_LIMIT => fn($meta) => $language->get('general', 'rate_limit', $meta),
```
Comment (supercrafter100) [inline modules/Core/pages/login.php:90]: Wouldn’t this have to be $meta->expires ?

Comment (supercrafter100) [reply modules/Core/pages/login.php:90]: Nvm

File: core/classes/Core/Validate.php [hunk 2]
```diff
@@ -318,6 +324,50 @@ public static function check(array $source, array $items = []): Validate {
...
+                        } else if (is_int($rule_value)) {
+                            // If integer default seconds to 60
+                            [$limit, $seconds] = [$rule_value, 60];
+                        }
+
+                        if (!isset($limit) || !isset($seconds)) {
+                            throw new Exception('Invalid rate limit configuration');
+                        }
+
+                        $session = $_SESSION["rate_limit_{$item}"];
```
Comment (Aberdeener) [inline core/classes/Core/Validate.php:341]: We are repeating the "rate_limit_{$item}" a few times here, can we create a variable like $rate_limit_key to not need to repeat it?

**Review by Aberdeener — CHANGES_REQUESTED** [review]

**Review by partydragen — APPROVED** [review]

**Review by Aberdeener — APPROVED** [review]
Great work, this is nice to have! Will try to remember to use it in places we have other cooldown systems right now

**Pull request merged**